### PR TITLE
feat: stack component + responsive props

### PR DESCRIPTION
**Pull request by kylemcd**
`@telegraph/layout`
- Adds Stack component
- Adds responsive props to layout component values
- Modularize and co-locate helper functions

`@telegraph/postcss-config`
- Adds custom media query plugin

### CODE REVIEW

File: examples/playground/app/page.tsx
Comment (kylemcd) [inline examples/playground/app/page.tsx:11]: Example of responsive values in action.

File: packages/layout/src/Box/Box.styles.css
Comment (kylemcd) [inline packages/layout/src/Box/Box.styles.css:37]: This sets the css variable to the correct value based on the screen size.

File: packages/layout/src/Box/Box.tsx
Comment (kylemcd) [inline packages/layout/src/Box/Box.tsx:51]: This functionality is moved into the helpers folder.

File: packages/layout/src/helpers/breakpoints.ts
Comment (kylemcd) [inline packages/layout/src/helpers/breakpoints.ts:7]: This function gets values for each breakpoint based on the object passed in. This also fills out values for breakpoints that were not passed in. 

For example
```
{ md: "20", xxl: "2" }
```

If this object is passed in sm-md are set to `20`, and lg-xxl are set to `2`. This works the same way as min width media queries work in css.

File: packages/tokens/scripts/generate-css-file.js
Comment (kylemcd) [inline packages/tokens/scripts/generate-css-file.js:75]: Creates a `breakpoint.css` file that we then use in the `postcss-config` in order to be able to utilize css variable syntax with `@custom-media`.

File: packages/tokens/src/tokens/breakpoint.ts
Comment (kylemcd) [inline packages/tokens/src/tokens/breakpoint.ts:6]: Would it make more sense to make these something like "phone" "tablet" "desktop" or the like?

File: packages/layout/src/helpers/breakpoints.ts
Comment (thomaswhyyou) [inline packages/layout/src/helpers/breakpoints.ts:1]: Looking at some of these helper functions made me wonder if there's any merit to writing some simple tests to verify / strap them in? Not suggesting we must btw, more of a thought that occurred to me as i'm looking over, and whether they would be helpful to us and for the future - especially since i don't necessarily grok everything they are doing from a glance.

**Review by thomaswhyyou — COMMENTED** [review]
Left a few comments/questions but not a ton of notes though, I assume they are pretty straight forward utility components that should do what you think they will do 😄

File: packages/tokens/src/tokens/breakpoint.ts
Comment (cjbell) [reply packages/tokens/src/tokens/breakpoint.ts:6]: Plus one here to the sizes you have here and to using `2xl` over `xxl`!

File: packages/tokens/src/tokens/breakpoint.ts
Comment (thomaswhyyou) [inline packages/tokens/src/tokens/breakpoint.ts:2]: One thing I meant to ask: where are these breakpoints coming from? Did we grab them from another framework defaults? Or is this our own definition?

File: packages/tokens/src/tokens/breakpoint.ts
Comment (kylemcd) [reply packages/tokens/src/tokens/breakpoint.ts:6]: Cool this works!

File: packages/layout/src/helpers/css-variables.ts
Comment (kylemcd) [reply packages/layout/src/helpers/css-variables.ts:37]: Yeah, what do you think would be a more straightforward name? `props` are the actual props passed into the component where as `propsMap` is the mapping of prop info in the constants file.

File: packages/tokens/src/tokens/breakpoint.ts
Comment (kylemcd) [reply packages/tokens/src/tokens/breakpoint.ts:2]: Didn't think to pull from a framework. I've just used these breakpoints since the dawn of time. Maybe it makes sense to grab the ones defined here by tailwind instead? https://tailwindcss.com/docs/screens

File: packages/layout/src/helpers/breakpoints.ts
Comment (kylemcd) [reply packages/layout/src/helpers/breakpoints.ts:1]: Yeah I was going to write tests for all of these once we get functionality finalized!

File: packages/layout/src/Stack/Stack.tsx
Comment (kylemcd) [reply packages/layout/src/Stack/Stack.tsx:13]: It's meant to be optional with a default of zero, missed the `?` when I renamed it.

File: packages/tokens/scripts/generate-css-file.js
Comment (kylemcd) [reply packages/tokens/scripts/generate-css-file.js:75]: Yeah I can change the css file name. The idea for the tokens themselves is that they're singular, so you can reference `breakpoint-sm` instead of `breakpoints-sm`

File: packages/layout/src/Box/Box.tsx
Comment (thomaswhyyou) [reply packages/layout/src/Box/Box.tsx:16]: > like a box with a gray rounded border, would be added as variants here.

Hmm, that sounds kinda strange to me for Box, and maybe worth a chat at some point. But not blocking this PR.

File: packages/tokens/src/tokens/breakpoint.ts
Comment (thomaswhyyou) [reply packages/tokens/src/tokens/breakpoint.ts:2]: > Maybe it makes sense to grab the ones defined here by tailwind instead? https://tailwindcss.com/docs/screens

Yeah that seems like a pretty sensible choice / starting place? And document where that came from for completeness.

File: packages/layout/src/Stack/Stack.tsx
Comment (thomaswhyyou) [inline packages/layout/src/Stack/Stack.tsx:28]:
```suggestion
    // Filter out the stack props from the rest of the props
```

File: packages/layout/src/Box/Box.tsx
Comment (kylemcd) [reply packages/layout/src/Box/Box.tsx:16]: Sounds good!

File: packages/layout/src/helpers/css-variables.ts
Comment (thomaswhyyou) [reply packages/layout/src/helpers/css-variables.ts:37]: Ah I seeee. Maybe `props` and `mappings`? Or maybe i'm overthinking here, because I initially thought the "map" in "propsMap" was referring to the data type i.e. Record, but now I see it's referring to the mapping. 

The name of this arg isn't super consequential at the end of the day, so I will defer the final call to you. I think fine to leave it be if we can't think of a better name.

File: packages/tokens/src/tokens/breakpoint.ts
Comment (thomaswhyyou) [inline packages/tokens/src/tokens/breakpoint.ts:1]: Worth leaving a note here to say these are referenced from tailwind defaults?

**Review by thomaswhyyou — APPROVED** [review]
👏

**Pull request merged**
